### PR TITLE
Add run stats.

### PR DESCRIPTION
**Pull request by MitjaBezensek**
Fixes https://github.com/ds300/lazyrepo/issues/18

Need to check why tests are failing, seeing some whitespace issues, but only in the dependent tests 😂

### CODE REVIEW

File: src/commands/run.js [hunk 1]
```diff
@@ -43,12 +44,18 @@ export async function run({ taskName, options }) {
...
   const failedTasks = tasks.allFailedTasks()
-
   if (failedTasks.length > 0) {
     logger.fail(`Failed tasks: ${failedTasks.join(', ')}`)
   }
 
-  if (Object.entries(tasks.allTasks).every(([, task]) => task.status === 'success:lazy')) {
-    logger.log('\n' + rainbow('>>> MAXIMUM LAZY'))
+  const stats = tasks.getTaskStats()
+  logger.log(`\n\t   Tasks:\t${kleur.green(stats.successful)} successful, ${stats.allTasks} Total`)
```
Comment (MitjaBezensek) [inline src/commands/run.js:52]: Doesn't look like we support fixed width outputs. Could improve this if we change the logger we use.

Comment (MitjaBezensek) [reply src/commands/run.js]: Added `padString` function to help with aligning things.

**Review by ds300 — APPROVED** [review]
Thanks Mitja! Looking great :) I had some ideas for improvements but feel free to merge as-is and we can tweak later.

File: src/commands/run.js [hunk 2]
```diff
@@ -43,12 +45,22 @@ export async function run({ taskName, options }) {
...
+      stats.allTasks
+    } Total`,
+  )
+  logger.log(`${padString('Cached:')} ${stats['success:lazy']} cached, ${stats.allTasks} Total`)
+  if (stats.allTasks === stats['success:lazy']) {
+    logger.log(`${padString('Laziness:')} ${rainbow('>>> MAXIMUM LAZY')}`)
+  } else {
+    const laziness = Math.round((stats['success:lazy'] / stats.allTasks) * 100)
+    const color = laziness > 50 ? kleur.green : laziness > 25 ? kleur.yellow : kleur.red
+    logger.log(`${padString('Laziness:')} ${color(laziness.toString() + '%')}`)
```
[the remark this answers is not in the document]
Comment (MitjaBezensek) [reply src/commands/run.js:64]: Yeah, makes sense, had a bit of fun with this 😂

File: src/commands/run.js [hunk 3]
```diff
@@ -43,12 +45,27 @@ export async function run({ taskName, options }) {
...
+  const successOutput = `${kleur.green(stats.successful.toString() + ' successful')}, ${
+    stats.allTasks
+  } total`
+
+  let output = ''
+  if (stats['success:lazy'] === stats.allTasks) {
+    output = dedent`
+    
+          Tasks:     ${successOutput}
+         Cached:     ${rainbow('>>> MAXIMUM LAZY')}`
```
Comment (MitjaBezensek) [inline src/commands/run.js:62]: Still uses `Cached:` prefix. Felt I could not get the alignment right without it, but we can drop it if you'd like.

File: src/commands/run.js [hunk 4]
```diff
@@ -43,12 +45,23 @@ export async function run({ taskName, options }) {
...
+
+  const cachedOutput =
+    stats.allTasks === stats['success:lazy']
+      ? rainbow('>>> MAXIMUM LAZY')
+      : `${stats['success:lazy']} cached, ${stats.allTasks} total`
+
+  const output = dedent`
+    
+          Tasks:     ${successOutput}
+         Cached:     ${cachedOutput}`
```
Comment (MitjaBezensek) [inline src/commands/run.js:65]: Using `Cached:` prefix even for the `>>> MAXIMUM LAZY`. I tried a few positions without it and the alignment just looked wrong to me, no matter what I tried. Feel free to drop it if you'd like though.

File: package.json [hunk 5]
```diff
@@ -54,6 +54,7 @@
     "slice-ansi": "^6.0.0",
     "source-map-support": "^0.5.21",
     "strip-ansi": "^7.0.1",
+    "ts-dedent": "^2.2.0",
```
Comment (MitjaBezensek) [inline package.json:57]: `dedent` seems unmaintained and had some issues with types so I used `ts-dedent`.

File: src/execCli.js [hunk 6]
```diff
@@ -72,7 +73,7 @@ export async function execCli(argv) {
   try {
     cli.parse(argv, { run: false })
     await cli.runMatchedCommand()
-    logger.success(`Done in ${timer.formatElapsedTime()}`)
+    logger.log(`${padString('Time:')} ${timer.formatElapsedTime()}\n`)
```
Comment (ds300) [inline src/execCli.js:76]: let's remove this and the `padString` function, then move the timer and the `Time:` label into the `run` command. It's not so useful for the other tasks.

Comment (MitjaBezensek) [reply src/execCli.js]: Done 👍

**Comment by ds300** [comment]
woohoo thanks ❤️

**Comment by MitjaBezensek** [comment]
🚀
